sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/common/Counter.java

```diff
@@ -105,11 +105,18 @@
    * @return the newly constructed Counter
    * @throws IllegalArgumentException if the aggregation kind is not supported
    */
-  public static Counter<Integer> ints(String name, AggregationKind kind) {
+  public static Counter<Integer> ints(CounterNameAndMetadata name, AggregationKind kind) {
     return new IntegerCounter(name, kind);
   }
 
   /**
+   * Constructs a new {@code Counter<Integer>} with an unstructured name.
+   */
+  public static Counter<Integer> ints(String name, AggregationKind kind) {
```

Comment:
I think this is for backwards compatibility? Should we mark this `@Deprecated`? Should it just be removed?